docs/angular/rxobject.md

```diff
@@ -0,0 +1,416 @@
+
+```ts
+let subject: Subject<string> = new ReplaySubject(1);
+let value$: Observable<string> = subject.asObservable();
+
+// Access latest data from the subject
+value$.subscribe((stringValue) => {
+  console.log(stringValue);
+})
+```
+
+As you can see, the code above creates an `Observable` from the `ReplaySubject` which we can subscribe to!
+
+Now that you have reviewed the basics of RxJS and its classes, we can now begin to bundle these classes together to create `RxObject` - a unified way for us to dynamically update our data.
+
+## Introduction to the `RxObject` Class
+
+The motivation of the `RxObject` class is to allow for the updating of data within observables such that your Angular Components can dynamically update with the new data. The `RxObject` class is not part of RxJS; rather, it is custom to the CSXL Web Application and serves as a helper that bundles together RxJS functionality in a concise and simple way. The class already exists in the CSXL repository at `frontend/src/app/rx-object.ts`. However in this section, we are going to rebuild this class from scratch so that you gain a fundamental understanding of how it works.
+
+First, we know that `RxObject` must be flexible enough to handle various types of data, in the same way that `Observable` and `ReplaySubject` do. To enable this functionality, we will specify that `RxObject` takes a *generic type* `T`. We can do this in the class's header:
+
+```ts
+export abstract class RxObject<T> {
+  // Implementation here
+}
+```
+
+> **NOTE:** We use the keyword `export` on this class so that other TypeScript classes in our file can access this.
+
+Notice that we also make this class **`abstract`**. The core motivation behind in the `RxObject<T>` class is that we *do not instantiate it directly*. Instead, we want to have *subclasses* ***extend*** this base class for related usecases. For example, if we wanted to dynamically store data for organizations, we may want to create a subclass `RxOrganizationsList` that *extends* `RxObject<Organization[]>`. We will utilize this in the future, but for now, we are just working on the conceptual `abstract` class.
+
+From here, we know we need to successfully implement the Observer design pattern. We know that the Angular Components are going to be the *observers*, which means the only thing missing here is the *subject*. We want our `RxObject<T>` class to contain our subject.
+
+You can recall that, after examining the various options for our RxJS subject object, we decided that `ReplaySubject<T>` is the best for our use case. This is because the `ReplaySubject<T>` re-emits the latest value that is has to *new subscribers*. We will use a `ReplaySubject<T>` object to serve as the main *subject* to satisfy the requirements of the Observer design pattern.
```

Comment:
A key idea here is the pattern of: if we have loaded a piece of data before, let's serve that up from cache and then behind the scenes we can fire off a request for updated data and when it comes in we can emit the newer data if there are any changes.